File: src/client/hoard-ui/src/api/dtos/PortfolioPositionDto.ts
import type { PositionPerformanceDto } from "./PositionPerformanceDto";

export interface PortfolioPositionDto {
  instrumentId: number;
  instrumentName: string;
  instrumentTicker: string;

  openDate: string;      // ISO date: YYYY-MM-DD
  closeDate: string | null;

  performance: PositionPerformanceDto | null;

  portfolioPercentage: number;
}

File: src/server/Hoard.Core/Application/Portfolios/PortfolioPositionDto.cs
namespace Hoard.Core.Application.Portfolios;

public class PortfolioPositionDto
{
    public int InstrumentId { get; set; }
    public required string InstrumentName { get; set; }
    public required string InstrumentTicker { get; set; }
    
    public DateOnly OpenDate { get; set; }
    public DateOnly? CloseDate { get; set; }
    
    public PositionPerformanceDto? Performance { get; set; }
    
    public decimal PortfolioPercentage { get; set; }
}